src/Tools/dotnet-monitor/Extensibility/FolderExtensionRepository.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+using System.Globalization;
+using System.IO;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Extensibility
+{
+    internal class FolderExtensionRepository : ExtensionRepository
+    {
+        private const string ExtensionDefinitionFile = "extension.jsonc";
+        private readonly string _targetFolder;
+        private readonly IFileProvider _fileSystem;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public FolderExtensionRepository(IFileProvider fileSystem, ILoggerFactory loggerFactory, int resolvePriority, string targetFolder)
+            : base(resolvePriority, string.Format(CultureInfo.CurrentCulture, Strings.Message_FolderExtensionRepoName, targetFolder))
+        {
+            _fileSystem = fileSystem;
+
+            _targetFolder = targetFolder;
+
+            _loggerFactory = loggerFactory;
+        }
+
+        public override bool TryFindExtension(string extensionMoniker, out IExtension extension)
+        {
+            IFileInfo extensionDir = _fileSystem.GetFileInfo(extensionMoniker);
+
+            if (extensionDir.Exists && extensionDir.IsDirectory)
```

Comment:
If you are not specially handling the negation of this case (e.g. logging when it doesn't exist), then it shouldn't be necessary because it'll be checked with the combined path below.

Reply:
For both your `MissingDirectoryProvider` and the `PhysicalFileProvider`, it will not throw when calling `GetFileInfo` with a directory that does not exist.

Reply:
This is all gone now.